16.0 mig contract payment auto

9184Lucky opened the pull request.

lasley committed:
[ADD] contract_auto: Automatic Payment For Contracts

pedrobaeza committed:
[FIX] contract_payment_auto: Switch test to HttpCase

pedrobaeza committed:
[FIX] contract_payment_auto: Fix RST syntax

moylop260 committed:
[FIX] contract_payment_auto: Fix freze wkhtmltopdf freezed

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

fcayre committed:
[FIX] contract_payment_auto: transaction create must always get a token (#167)

When a contrat had no payment token but the corresponding partner had
one, the transaction was created without an acquirer, leading to an
integrity error in postgres.

This change makes sure the token used to test the ability to pay an
invoice is passed along to the transaction creation call.

Tests were also added to check the ability to use the contract token if
present, but the partner's in the opposite case.

This change fixes #165.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update contract_payment_auto.pot

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (25 of 25 strings)

Translation: contract-10.0/contract-10.0-contract_payment_auto
Translate-URL: https://translation.odoo-community.org/projects/contract-10-0/contract-10-0-contract_payment_auto/de/

OCA-git-bot committed:
[ADD] icon.png

norlinhenrik committed:
[MIG] contract_payment_auto: Migration to 12.0

norlinhenrik committed:
[MIG][FIX] contract_payment_auto: Migration to 12.0

norlinhenrik committed:
[MIG][FIX2] contract_payment_auto: Migration to 12.0

norlinhenrik committed:
[MIG][FIX3] contract_payment_auto: Migration to 12.0

fcayre committed:
[FIX] Make payment work with several invoices and contracts to be paid

The `_recurring_create_invoice` contract method returns multiple
invoices potentially coming from different contracts, which may have
different values for their `is_auto_pay` attribute.

Without this patch, this case simply crashes (a unit test with more than
one contract to be invoiced is a must-have in this module!).

fcayre committed:
[IMP] Fix small docstring typo

fcayre committed:
Fix tests

Change the mock method to patch the Contract class instead of the contract object:
patching the object is less reliable and depends on the implementation too much.
This implementation changed and the tests were no more working. Mocking the class
method fixes the problem.

fcayre committed:
Reactivate all tests and fix them

The payment acquirer API requires an implementation of a _get_feature_support
method. This is done using a temporary mock where required.

fcayre committed:
Do not crash if generated invoice has no line

fcayre committed:
[FIX] Add invoice on created payment transactions

Also pass the invoice to the transaction reference computation mecanism
so that it reflects the invoice's number.

Commit:
[IMP] contract_payment_auto: black, isort, prettier

Commit:
[MIG] contract_payment_auto: Migration to 16.0

Commit:
Change account.invoice to account.move

Commit:
Addapt to changes in the payment module.

Main changes are the change of payment.acquirer model to
payment.provider and disparition of s2s_do_transaction method.

Commit:
Modify test class decorator

Commit:
Replace model mock by model injection

This allows a cleaner overriding of _send_payment_request and imporve
code readability.

Commit:
Improve test readability

Honeyxilia commented on contract_payment_auto/tests/test_contract.py:
```diff
@@ -0,0 +1,328 @@
+        cls.loader.restore_registry()
+        super().tearDownClass()
+
+    def setUp(self):
```
A part si le module fonctionne d'une manière à modifier les variables créées ici, je pense que certaines des attributions (partner, product) ici peuvent être déplacées dans setUpClass, car uniquement `self.contract` est changé dans quelques tests

9184Lucky replied:
C'est réglé, j'ai tout bougé dans le setUpClass !

Honeyxilia reviewed (commented):
Pas beaucoup de remarques c:

Commit:
Move setUp to setUpClass to improve efficiency

Commit:
Make test class depend on TestContractBase

Initial purpose was to pass tests on a database with
l10n module installed but it also simplifies the setUpClass method.

Commit:
Modify mail template from jinja to qweb.

Also adapt fields and methods

fcayre commented on contract_payment_auto/tests/models.py:
```diff
@@ -0,0 +1,11 @@
+from odoo import models
+
+
+class TransactionTest(models.Model):
```
Le titre du commit je l'aurais plutôt écrit "Replace model mock by model injection", ça me paraît plus juste.

9184Lucky replied:
Fait

fcayre commented on contract_payment_auto/tests/test_contract.py:
```diff
@@ -10,11 +9,12 @@
 
 @tagged("-at_install", "post_install")
-class TestContract(HttpCase):
+class TestContract(HttpCase, TestContractBase):
```
Typos dans le message de commit : pas de s à depend et 1 ou 2 typos dans le corps
```Make test class depend on TestContractBase

Initial purpose was to pass tests on a database with
l10n module installed but it also simplifies the setUpClass method.
```

9184Lucky replied:
Fait

fcayre commented on contract_payment_auto/data/mail_template_data.xml:
```diff
@@ -9,41 +9,41 @@
-    % if object.partner_id.parent_id:
-        (<i>${object.partner_id.parent_id.name}</i>)
-    % endif
+    Hello <t t-out="object.partner_id.name"></t>
```
tag auto-fermant ?

9184Lucky replied:
Yes je vais reprendre ça mais à mon avis c'est une "faiblesse" du script de migration de mail, à mon avis pas trop compliqué d'améliorer pour traiter ça automatiquement !

fcayre commented on contract_payment_auto/data/mail_template_data.xml:
```diff
@@ -9,41 +9,41 @@
+    <a t-attf-href="{{access_url}}">
         <strong>
-            ${object.number}
+            <t t-out="object.number"></t>
```
tag auto-fermant plutôt ? (pas important)

fcayre commented on contract_payment_auto/data/mail_template_data.xml:
```diff
@@ -9,41 +9,41 @@
-            (with reference: ${object.origin} )
-        % endif
+        <t t-if="object.origin">
+            (with reference: <t t-out="object.origin"></t> )
```
idem

fcayre commented on contract_payment_auto/data/mail_template_data.xml:
```diff
@@ -61,41 +61,41 @@
-    % if object.partner_id.parent_id:
-        (<i>${object.partner_id.parent_id.name}</i>)
-    % endif
+    Hello <t t-out="object.partner_id.name"></t>
```
idem (je les note plus désormais, y en a d'autres).

fcayre commented on contract_payment_auto/data/mail_template_data.xml:
```diff
@@ -12,22 +12,22 @@
         >{{(object.user_id.email and '%s &lt;%s&gt;' % (object.user_id.name, object.user_id.email) or '')}}</field>
         <field
             name="subject"
-        >Automatic Payment Failure (Ref {{object.number or 'n/a'}})</field>
```
Bien vu !!!

fcayre commented:
/ocabot rebase

fcayre commented:
/ocabot merge patch

fcayre committed:
Add Commown SCIC as an author (required by pre-commit)

fcayre commented:
Fermée au profit de la PR ouverte auprès de l'OCA.
